core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1069,7 +1070,7 @@ object GroupMetadataManager {
    * @return key for offset commit message
    */
   def offsetCommitKey(groupId: String, topicPartition: TopicPartition): Array[Byte] = {
-    MessageUtil.toVersionPrefixedBytes(OffsetCommitKey.HIGHEST_SUPPORTED_VERSION,
+    MessageUtil.toVersionPrefixedBytes(CoordinatorRecordType.OFFSET_COMMIT.id(),
```

Comment:
it doesn't seem like this change alters the schema written/read to/from the log, but i want to confirm that this doesn't break compatibility. should we have a test for this? my question is if there is a bug in one of the record format changes, would that be caught somewhere?

also, internal coordinator topics (txn state) can use this but the difference would be to use a unique api key instead now, right?

Reply:
> it doesn't seem like this change alters the schema written/read to/from the log, but i want to confirm that this doesn't break compatibility. should we have a test for this? my question is if there is a bug in one of the record format changes, would that be caught somewhere?

Let me see if I can some unit tests for this. Otherwise, any regressions should also be caught by upgrade system tests.

> also, internal coordinator topics (txn state) can use this but the difference would be to use a unique api key instead now, right?

Right.

Reply:
btw, I did not change the coordinator tests so they still used hard coded record types. This also ensures that we don't introduce wrong ones.

Reply:
Added unit tests for the old group coordinator too. The new one has already good coverage (e.g. GroupCoordinatorRecordSerdeTest).

Reply:
> Added unit tests for the old group coordinator too.

could you point me to where it was added?

Reply:
core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala